packages/realm/src/app-services/PushClient.ts

```diff
@@ -0,0 +1,50 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { BSON, assert, binding } from "../internal";
+/**
+ * Authentication provider where users identify using an API-key.
+ */
+export class PushClient {
+  /** @internal */
+  private user: binding.SyncUser;
+  /** @internal */
+  public internal: binding.PushClient;
+
+  /** @internal */
```

Comment:
I'm not sure we should mark the constructor as internal. That leads to no constructor being included in the .d.ts, which is equivalent to having an implicit `constructor()`: [playground](https://www.typescriptlang.org/play?stripInternal=true#code/MYGwhgzhAEBiD29oG8BQBfVqB2BTA7nIgBQCUA3FqJDAEJgBOa0L0A9AFQfQACAltgAuuBtjAhoHNqlbRg8bBEEMArsEHwGxMAC5o2FQFsARiNIpMmHAWj0tFdm2iCAFnxj4FAckHQRDTWg+ADNoAE94FWgAKxUlaBBEAGtoMF9XXGgAOgATLMEIIA)

I'll file a bug with TS because there _should_ be a way to have a type that consumers can't construct.

Reply:
Bug filed: https://github.com/microsoft/TypeScript/issues/51486